dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -621,11 +636,155 @@ bool TransportRegistry::has_type(const TransportType_rch& type) const
   return type_map_.count(name);
 }
 
-bool TransportRegistry::has_transport_template() const
+bool TransportRegistry::create_new_transport_instance_for_participant(DDS::DomainId_t id, OPENDDS_STRING& config_name)
 {
-  return transport_templates_.size() > 0;
+  TransportConfig_rch cfg = get_config(domain_config_map_[id]);
+
+  OPENDDS_STRING inst_name = cfg->instances_[0]->name() + "_" + config_name;
```

Comment:
Are we sure domain_config_map_ will always have an entry for "id" and we're sure that it will always have at least one instance in it? Otherwise maybe have some sanity checks here to return false if those conditions aren't true.